compiler/src/iree/compiler/Codegen/Dialect/Codegen/IR/UKernelOps.cpp

```diff
@@ -28,6 +28,28 @@ namespace mlir::iree_compiler::IREE::Codegen {
 // Helpers
 //===---------------------------------------------------------------------===//
 
+static void getEffectsImpl(
+    SmallVectorImpl<SideEffects::EffectInstance<MemoryEffects::Effect>>
+        &effects,
+    ValueRange inputOperands, ValueRange outputOperands) {
+  for (Value value : inputOperands) {
+    if (!llvm::isa<MemRefType>(value.getType())) {
+      continue;
+    }
+    effects.emplace_back(MemoryEffects::Read::get(), value,
+                         SideEffects::DefaultResource::get());
+  }
+  for (Value value : outputOperands) {
```

Comment:
You also need to look at `other_operands` and add them as read semantics.

Reply:
good catch, thanks!